### PR TITLE
make.bat: Don't override SPHINXOPTS/O from the environment

### LINKED ISSUE #23622: [MNT]: make.bat not parsing sphinxopt
### Summary

passing arguments to sphinx doesn't work, but [the documented way](https://matplotlib.org/devdocs/devel/documenting_mpl.html#documenting-matplotlib) is supposed to. Pretty sure it's that https://github.com/matplotlib/matplotlib/blob/a68f21f589b1747cccf30d2f9bd460bacf397003/doc/make.bat#L13 overwrites any input. Checked via

```cmd
> set SPHINXOPTS=-j4 -Dplot_gallery=0
> echo %SPHINXOPTS%
-j4 -Dplot_gallery=0
> make html
> echo %SPHINXOPTS%
-W
```
and yeah I tried `` and no quote versions

### Proposed fix

Amend `make.bat` so that it parses and correctly sets `SPHINXOPTS` arguments. <s>This is a medium good first issue in that it's self contained, but requires knowing how to write .bat files. </s>

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

I have not actually tested this... cc @story645

`if defined` does require command extensions to be enabled, but a) they're enabled by default, and b) we need them for `%~dp0` above already.

Fixes #23622 

## PR Checklist

**Tests and Styling**
- [n/a] Has pytest style unit tests (and `pytest` passes).
- [n/a] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [n/a] New features are documented, with examples if plot related.
- [n/a] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [n/a] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [n/a] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: doc/make.bat [hunk 1]
```diff
@@ -10,8 +10,9 @@ if "%SPHINXBUILD%" == "" (
 set SOURCEDIR=.
 set BUILDDIR=build
 set SPHINXPROJ=matplotlib
+if defined SPHINXOPTS goto skipopts
 set SPHINXOPTS=-W
```
Comment (story645) [inline doc/make.bat:14]: if we'
```suggestion
set SPHINXOPTS=-W --keep-building
```
that's what the docs say is the default and that's what's set in the makefile

**Review by story645 — CHANGES_REQUESTED** [review]

**Comment by story645** [comment]
this seems to work and is I think more direct:
```bat
if not defined SPHINXOPTS (
set SPHINXOPTS=-W --keep-going
)
```
but in both versions `SPHINXOPTS=` is overwritten, but `SPHINXOPTS=''" isn't so we should probably just put that in the docs?

**Comment by QuLogic** [comment]
I didn't use `not defined` because that didn't appear to be specified in the docs: https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/if

Does `set SPHINXOPTS=` set a variable empty, or undefine it in Windows? Are you setting it in the environment or passing it at the end of the `make.bat` call?

**Comment by story645** [comment]
not is specified as optionally usable with a parameter and I haven't found anything that says it wouldn't work,  but also I won't block on that🤷‍♀️ 
> if [not] ERRORLEVEL <number> <command> [else <expression>]

> Does set SPHINXOPTS= set a variable empty, or undefine it in Windows? 

I think undefines it: 

```bat
>set SPHINXOPTS=        
>echo %SPHINXOPTS%
%SPHINXOPTS%
```

>Are you setting it in the environment or passing it at the end of the make.bat call? 

tried both

**Comment by story645** [comment]
Ok so putting at end doesn't work, put in a quick PR so docs reflect current state of build #23640 but once this PR works I think the docs get amended such that https://github.com/matplotlib/matplotlib/blob/9f5c041158024733c68fbc15eaee57cbe97fdab7/doc/devel/documenting_mpl.rst?plain=1#L100 becomes
```
> On Windows, set the options as environment variables, e.g.:
```bat
set O=-W --keep-going -j4 & make html
```

**Review by story645 — APPROVED** [review]
the .bat file seems to work now

**Comment by QuLogic** [comment]
I don't mind committing here, just not the default commit message.

**Pull request merged**